### PR TITLE
feat: add a way to choose platform when building image

### LINKED ISSUE #5492: allow to provide platform parameter when doing builds
### Is your enhancement related to a problem? Please describe

When building image it is now providing the default platform (the one used by the podman machine)

we should be able to select the platform

### Describe the solution you'd like

allow to select the platform

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Allow to build for a different platform when building image within Podman Desktop

### Screenshot / video of UI

image

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/5492

### How to test this PR?

unit tests added

you can also go to build panel and choose a platform

### CODE REVIEW

File: packages/renderer/src/lib/image/BuildImageFromContainerfileCard.svelte [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    {#if !additionalItem}
+      <Checkbox bind:checked="{checked}" on:click="{() => handleClick()}" />
+    {:else}
+      <div>&nbsp;</div>
+    {/if}
+  </div>
+  <div class="mr-2 text-gray-700">
+    <div class="flex flex-row">
+      <div class="ml-1">
+        {#if fontAwesomeIcon}
```
Comment (deboer-tim) [inline packages/renderer/src/lib/image/BuildImageFromContainerfileCard.svelte]: Could just use 'startsWith('fa')' since we essentially do this elsewhere.

**Review by deboer-tim — APPROVED** [review]
Code LGTM, and nice improvement to have. My only concern is that the architecture cards are large and very noticeable - i.e. drawing your eyes and putting focus on an option that we only expect to be used in some cases. I think I'd prefer if they were closer to badges (e.g. maybe just icon + amd) and you used tooltips if you didn't recognize it. But, we could always do this later.

**Comment by benoitf** [comment]
cc @mairin for design review

File: packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    value: item.value,
+    icon: faLayerGroup,
+    fontAwesomeIcon: true,
+  };
+  advancedCards.push(card);
+  advancedCards = advancedCards;
+}
+</script>
+
+<div class="flex flex-col">
```
Comment (odockal) [inline packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte]: I would consider also adding ARIA functionality from the start of this PR. 
adding `role="region" aria-label="Build Platform Options"` or something similar would be nice.

File: packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+  <div class="flex flex-row gap-x-4 gap-y-4 flex-wrap">
+    {#each sortedCards as card}
+      <BuildImageFromContainerfileCard
+        title="{card.title}"
+        isDefault="{card.isDefault}"
+        checked="{card.checked}"
+        badge="{card.badge}"
+        value="{card.value}"
+        icon="{card.icon}"
```
Comment (odockal) [inline packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte:137]: `aria-label="{card.value}"`

File: packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    <div class="flex flex-col pt-4">
+      <div class="flex flex-row gap-x-4 flex-wrap gap-y-4">
+        {#each advancedCards as card}
+          <BuildImageFromContainerfileCard
+            title="{card.title}"
+            isDefault="{card.isDefault}"
+            checked="{card.checked}"
+            badge="{card.badge}"
+            value="{card.value}"
+            icon="{card.icon}"
```
Comment (odockal) [inline packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte:160]: `aria-label="{card.value}"`

File: packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            fontAwesomeIcon="{card.fontAwesomeIcon}"
+            on:card="{item => handleCard(item)}" />
+        {/each}
+        <BuildImageFromContainerfileCard
+          title="New platform"
+          isDefault="{false}"
```
Comment (odockal) [inline packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte:165]: `aria-label="New Platform"`

**Review by odockal — CHANGES_REQUESTED** [review]

**Comment by benoitf** [comment]
@odockal while I understand the first aria stuff around `role="region" aria-label="Build Platform Options"` I don't understand the other one about aria-labels on Cards object.

The card element already have a div with the value

File: packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte (hunk 2 above)
Comment (benoitf) [reply packages/renderer/src/lib/image/BuildImageFromContainerfileCards.svelte]: added

File: packages/renderer/src/lib/image/BuildImageFromContainerfileCard.svelte (hunk 1 above)
Comment (benoitf) [reply packages/renderer/src/lib/image/BuildImageFromContainerfileCard.svelte]: I changed it.
I think in other places it's checking prefix being exactly 'fa' and for Linux icon as a brand it was 'fab'

**Review by odockal — APPROVED** [review]

**Comment by benoitf** [comment]
@deboer-tim discussed with @mairin and she'll provide a new UI based on your feedback

**Pull request merged**